evals/compare_original_and_generated.py

```diff
@@ -0,0 +1,220 @@
+
+    return dataloader
+
+
+def get_maestro_record(query: str | int):
+    ds = load_dataset("roszcz/maestro-v1-sustain", split="validation")
+    quantizer = MidiQuantizer(7, 7, 7)
+
+    if isinstance(query, int):
+        record = ds[query]
+    else:
+        idx_query = [i for i, record in enumerate(ds) if str.lower(query) in str.lower(f"{record['composer']} {record['title']}")]
+        # get first record
+        idx = idx_query[0]
+        record = ds[idx]
+
+    midi_filename = f"{record['composer']} {record['title']}"
+    piece = ff.MidiPiece.from_huggingface(record)
+
+    piece.df["next_start"] = piece.df.start.shift(-1)
+    piece.df["dstart"] = piece.df.next_start - piece.df.start
+
+    piece_quantized = quantizer.quantize_piece(piece)
+
+    sequence = {
+        "filename": midi_filename,
+        "pitch": torch.tensor(piece.df.pitch, dtype=torch.long)[:1024],
+        "dstart": torch.tensor(piece.df.dstart, dtype=torch.float)[:1024],
+        "duration": torch.tensor(piece.df.duration, dtype=torch.float)[:1024],
+        "velocity": (torch.tensor(piece.df.velocity, dtype=torch.float)[None, None, :1024] / 64) - 1,
+        "dstart_bin": torch.tensor(piece_quantized.df.dstart_bin, dtype=torch.long)[None, :1024],
+        "duration_bin": torch.tensor(piece_quantized.df.duration_bin, dtype=torch.long)[None, :1024],
+        "velocity_bin": torch.tensor(piece_quantized.df.velocity_bin, dtype=torch.long)[None, :1024],
+    }
+
+    return sequence
+
+
+def denormalize_velocity(velocity: np.ndarray):
+    return (velocity + 1) * 64
+
+
+def to_midi_piece(pitch: np.ndarray, dstart: np.ndarray, duration: np.ndarray, velocity: np.ndarray) -> ff.MidiPiece:
+    record = {
+        "pitch": pitch,
+        "velocity": velocity,
+        "dstart": dstart,
+        "duration": duration,
+    }
+
+    df = pd.DataFrame(record)
+    df["start"] = df.dstart.cumsum().shift(1).fillna(0)
+    df["end"] = df.start + df.duration
+
+    return ff.MidiPiece(df)
+
+
+def compare_original_and_generated(
+    gen: Generator,
+    gen_ema: Generator,
+    query: str | int,
+    cfg: OmegaConf,
+    conditioning_model: nn.Module = None,
+    classifier_free_guidance_scale: float = 3.0,
+):
+    # loader = preprocess_dataset("JasiekKaczmarczyk/maestro-sustain-quantized", batch_size, 1)
+
+    batch = get_maestro_record(query)
+
+    # grab only name without extension
+    filename = batch["filename"]
+
+    # unpack other attributes
+    pitch = batch["pitch"].numpy()
+    dstart = batch["dstart"].numpy()
+    duration = batch["duration"].numpy()
+    velocity = batch["velocity"]
+
+    velocity_bin = batch["velocity_bin"].to(cfg.train.device)
+    dstart_bin = batch["dstart_bin"].to(cfg.train.device)
+    duration_bin = batch["duration_bin"].to(cfg.train.device)
+
+    if conditioning_model is not None:
+        with torch.no_grad():
+            label_emb = conditioning_model(velocity_bin, dstart_bin, duration_bin)
```

Comment:
```suggestion
            conditioning_embedding = conditioning_model(velocity_bin, dstart_bin, duration_bin)
```

`label_emb` is really mysterious, let's change it everywhere to something more meaningful (I'm not sure if my suggestions is great)